src/linker/Linker/MessageOrigin.cs

```diff
@@ -75,5 +76,23 @@ public override string ToString ()
 		public override int GetHashCode () => (FileName, MemberDefinition, SourceLine, SourceColumn).GetHashCode ();
 		public static bool operator == (MessageOrigin lhs, MessageOrigin rhs) => lhs.Equals (rhs);
 		public static bool operator != (MessageOrigin lhs, MessageOrigin rhs) => !lhs.Equals (rhs);
+
+		public int CompareTo (MessageOrigin other)
+		{
+			if (MemberDefinition != null && other.MemberDefinition != null) {
+				return (MemberDefinition.DeclaringType.Module.Assembly.Name.Name, MemberDefinition.DeclaringType.Name, MemberDefinition.Name).CompareTo
+					((other.MemberDefinition.DeclaringType.Module.Assembly.Name.Name, other.MemberDefinition.DeclaringType.Name, other.MemberDefinition.Name));
```

Comment:
We should do `MemberDefinition?.DeclaringType?.Module?.Assembly?.Name?.Name` - this code should better work no matter what crazy input it gets - and with Cecil it's hard to rely on non-null anywhere.
Simply put - I would rather have wrong sorting than crash the linker while writing out warnings.